src/query/functions/src/scalars/array.rs

```diff
@@ -751,10 +751,12 @@ fn register_array_aggr(registry: &mut FunctionRegistry) {
                     }];
                     let columns = vec![BlockEntry{
                         data_type: arr.data_type(),
-                        value: Value::Column(arr)
+                        value: Value::Column(arr.clone()),
                     }];
-                    let sort_block = DataBlock::sort(&DataBlock::new(columns, len), &sort_desc, None).unwrap();
-                    sort_block.columns()[0].value.clone().into_column().unwrap()
+                    match DataBlock::sort(&DataBlock::new(columns, len), &sort_desc, None) {
```

Comment:
Ignoring errors may lead to unintended results. We need to implement sort for `Null`, `EmptyArray`, and `EmptyMap` types in `build_compare` function. Add a custom compare function for those types, for example
```rust
fn compare_null() -> ArrowResult<DynComparator> {
    Ok(Box::new(move |_, _| Ordering::Equal))
}
```

Reply:
i have completed this. Please review this pr again. Thank you very much.